facilities/views/facility_views.py

```diff
@@ -775,48 +784,92 @@ class RegulationStatusDetailView(
     serializer_class = RegulationStatusSerializer
 
 
-def get_inspection_report(request, facility_id):
-    facility = Facility.objects.get(pk=facility_id)
-    template = loader.get_template('inspection_report.txt')
-    report_date = timezone.now().isoformat()
-    regulating_bodies = FacilityRegulationStatus.objects.filter(
-        facility=facility)
-    regulating_body = regulating_bodies[0] if regulating_bodies else None
-
-    context = Context(
-        {
-            "report_date": report_date,
-            "facility": facility,
-            "regulating_body": regulating_body
-        }
-    )
-    return HttpResponse(template.render(context))
-
+class DownloadPDFMixin(object):
+    def download_file(self, doc, file_name):
+        doc_file_name = 'temp'
+        file_path = os.path.join(settings.BASE_DIR, file_name)
+        doc_file_path = os.path.join(settings.BASE_DIR, doc_file_name)
+        writting_file = open(doc_file_path, 'w')
```

Comment:
Does not work well with the weasyprint. The pdf generated using StringIO has errors.